internal/engine/build_and_deployer.go

```diff
@@ -14,4 +15,10 @@ type BuildAndDeployer interface {
 	// BuildResult can be used to construct a BuildState, which contains
 	// the last successful build and the files changed since that build.
 	BuildAndDeploy(ctx context.Context, service model.Service, currentState BuildState) (BuildResult, error)
+
+	// BaD needs to be able to get the container that a given build affected, so that
+	// it can do incremental builds on that container if needed.
+	// NOTE(maia): this isn't quite relevant to ImageBuildAndDeployer,
+	// consider putting elsewhere or renaming (`Warm`?).
+	GetContainerForBuild(ctx context.Context, build BuildResult) (k8s.ContainerID, error)
```

Comment:
i thought the original idea was to have this return a new BuildResult, rather than make the caller attach the ContainerID to the BuildResult, tho i guess this is ok too